assessments/projects/knative/self-assessment.md

```diff
@@ -0,0 +1,577 @@
+  - https://github.com/knative/serving/blob/main/config/README.md
+  - https://github.com/knative/eventing/tree/main/config
+
+### Software Bill of Materials
+
+- The go.mod files for each component can be found in the root directory of each repo, linked below. Knative-automation automatically updates the go.mod file with each deployment.
+  - https://github.com/knative/serving/blob/main/go.mod
+  - https://github.com/knative/func/blob/main/go.mod
+  - https://github.com/knative/eventing/blob/main/go.mod
+- Additionally, each version contains a list of each added or modified dependency in the version release notes, so one can see a history of the dependencies.
+  - https://github.com/knative/func/releases
+  - https://github.com/knative/serving/releases
+  - https://github.com/knative/eventing/releases
+- The knative/pkg repository is a storage site for common Knative packages across all other Knative repositories for each component.
+  - https://github.com/knative/pkg
+
+---
+
+## Overview
+Knative is an open-source platform that enhances Kubernetes, focusing on the efficient deployment and management of serverless, cloud-native applications. It introduces two main components: Serving and Eventing. Serving automates the scaling and management of serverless workloads, while Eventing deals with event-driven architecture, managing and triggering serverless functions. This integration with Kubernetes not only streamlines the deployment process but also boosts developer productivity and reduces operational costs. Knative's cloud-agnostic nature and extensibility make it a versatile choice for diverse serverless computing environments.
+Source: https://knative.dev/docs/
+### Background
+Historically, in the world of container orchestration and cloud-native development, managing and deploying serverless applications has been a complex task. Traditional methods often required developers to manually handle aspects like scaling, updating, and event management for each application separately. This approach was not only time-consuming but also prone to inconsistencies, especially in environments with a diverse set of applications and services.
+
+In cloud-native ecosystems, especially those using Kubernetes, the management of serverless workloads presented unique challenges. Kubernetes, while powerful, did not natively offer specialized tools for serverless architecture, leading developers to either rely on external tools or build custom solutions. These methods, though effective, lacked uniformity and efficiency, especially in handling auto-scaling, rapid deployment, and event-driven architecture.
+
+Knative emerged as a solution to these challenges. It was designed to simplify the process of deploying and managing serverless applications on Kubernetes. By providing middleware components, Knative enables developers to focus more on building their applications rather than worrying about the underlying infrastructure.
+
+### Actors
```

Comment:
This section is very detailed with several actors, and it's easy to get lost in it. 
Can you please focus add an architectural diagram which helps relate different components in the system, and limit the description to the knative actors ("individual parts of your system that interact to provide the desired functionality.") . In addition, actors only need to be separate, if they are isolated in some way.

Reply:
Do you think if we should include Functions in our actors?

Reply:
Yes, particularly since many of the components of functions run on the user's laptop, vs running on the cluster like the other components.